core/src/main/groovy/com/novoda/gradle/release/PropertyFinder.groovy

```diff
@@ -0,0 +1,39 @@
+package com.novoda.gradle.release
+
+import org.gradle.api.Project;
+
+class PropertyFinder {
+
+    private final Project project
+    private final PublishExtension extension
+
+    PropertyFinder(Project project, PublishExtension extension) {
+        this.extension = extension
+        this.project = project
+    }
+
+    def getBintrayUser() {
+        getString(project, 'bintrayUser', extension.bintrayUser)
+    }
+
+    def getBintrayKey() {
+        getString(project, 'bintrayKey', extension.bintrayKey)
+    }
+
+    def getDryRun() {
+        getBoolean(project, 'dryRun', extension.dryRun)
+    }
+
+    def getPublishVersion() {
+        getString(project, 'publishVersion', extension.publishVersion == null ? extension.version : extension.publishVersion)
```

Comment:
`getString(project, 'publishVersion', extension.publishVersion ?: extension.version)`

http://docs.groovy-lang.org/latest/html/documentation/index.html#_elvis_operator

The "Elvis operator" is a shortening of the ternary operator. Basically means if not null null return myself

> with the Elvis operator, the value which is tested is used if it is not false or null